src/modules/dirwatcher.js

```diff
@@ -0,0 +1,37 @@
+import EventEmitter from 'events'
+import fs from 'fs'
+import Set from 'collections/set'
+
+export default class DirWatcher extends EventEmitter {
+
+    constructor(path, delay) {
+        super();
+        this.allFiles = new Set();
+        this.watch(path, delay);
+    }
+
+    watch(path, delay) {
+        setInterval(() => {
+            fs.readdir(path, (error, files) => {
+                if (error) {
+                    console.log(error);
+                    throw error;
+                } else {
+                    if (this.allFiles.length == 0) {                        
+                        files.forEach(element => {
+                            this.allFiles.add(element);
+                        });                        
+                        this.emit("change", this.allFiles.map(filename => {return path + '/' + filename}));
```

Comment:
Usage of arrow functions usually considers that if you have only 1 LOC in it it may be transformed to `.map(filename => path + '/' + filename)` with explicit `return` statement